music_player/scripts/music_player_server.py

```diff
@@ -58,6 +59,8 @@ def __init__(self):
             outfile.close()
 
         self.file_names = [join(self.audio_folder, f[0]) for f in file_set]
+        if rospy.get_param("~shuffle", False):
+            random.shuffle(self.file_names)
```

Comment:
The node is started and stopped with every new walking group task. hence, the list should have a different order for every walking group. I could also shuffle the songs whenever all of them have been played but that would lead to illegible behaviour for the next and previous services and also might mean that the same song is played twice in a row if the song was at the end of the previous list and then got shuffled to the front of the new list.

Another possibility is to have a shuffle service but I don't think that is necessary since the music player is restarted every time any way.